tests/base/polynomials_integrated_legendre_sz.cc

```diff
@@ -0,0 +1,87 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// This tests the stability of the polynomial evaluation of IntegratedLegendreSZ.
+
+#include "../tests.h"
+#include <iomanip>
+#include <fstream>
+#include <cmath>
+
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/polynomial.h>
+#include <deal.II/base/polynomials_integrated_legendre_sz.h>
+#include <deal.II/base/quadrature_lib.h>
+
+
+using namespace Polynomials;
+
+
+void check_at_one (const std::vector<Polynomial<double> > &p)
+{
+  // Ignore first two polynomials as the integrated Legendre polynomials are only defined
+  // for degree > 1, it is only added to maintain the recursive relation.
+
+  deallog << "Function value of polynomial at right end point: ";
+  for (unsigned int i=2; i<p.size(); ++i)
+    {
+      deallog << '.';
+      const double y = p[i].value(1.);
+      if (std::fabs(y) > 1e-13)
+        deallog << "Error1  lg y=" << std::log10(std::fabs(y)) << std::endl;
```

Comment:
It's interesting that you get to the accuracy of `1e-13` in the coefficient form which is otherwise notoriously unstable at high degrees. I guess it is because 1 is a special point for the polynomials and there is not much cancellation. If we can, we try to express the polynomials in something we call *root form* which is stable for all degrees. (The evaluation is related to barycentric evaluation of Lagrange polynomials described in a number of books on high-order methods but our current interfaces in deal.II represent each polynomial separately.) But I would not see the roots of the integrated Legendre polynomials immediately, so I would keep things this way.